pkg/controller/mutators/stats_reporter_test.go

```diff
@@ -0,0 +1,83 @@
+package mutators
+
+import (
+	"testing"
+	"time"
+
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+func TestReportMutatorIngestionRequest(t *testing.T) {
+	expectedTags := map[string]string{
+		"status": "active",
+	}
+
+	const (
+		expectedDurationValueMin         = time.Duration(1 * time.Second)
+		expectedDurationValueMax         = time.Duration(5 * time.Second)
+		expectedDurationMin      float64 = 1
+		expectedDurationMax      float64 = 5
+		expectedRowLength                = 1
+	)
+
+	r := NewStatsReporter()
+	err := r.ReportMutatorIngestionRequest(MutatorStatusActive, expectedDurationValueMin)
+	if err != nil {
+		t.Errorf("ReportRequest error %v", err)
+	}
+	err = r.ReportMutatorIngestionRequest(MutatorStatusActive, expectedDurationValueMax)
+	if err != nil {
+		t.Errorf("ReportRequest error %v", err)
+	}
+
+	// Count test
+	row := checkData(t, mutatorIngestionCountMetricName, expectedRowLength)
+	count, ok := row.Data.(*view.CountData)
+	if !ok {
+		t.Error("ReportRequest should have aggregation Count()")
+	}
+	if count.Value != 2 {
+		t.Errorf("Metric: %v - Expected %v, got %v. ", mutatorIngestionCountMetricName, 2, count.Value)
+	}
+
+	verifyTags(t, expectedTags, row.Tags)
+
+	// Duration test
+	row = checkData(t, mutatorIngestionDurationMetricName, expectedRowLength)
+	durationValue, ok := row.Data.(*view.DistributionData)
+	if !ok {
+		t.Error("ReportRequest should have aggregation Distribution()")
+	}
+	if durationValue.Min != expectedDurationMin {
+		t.Errorf("got tag '%v' min %v, want %v", mutatorIngestionDurationMetricName, durationValue.Min, expectedDurationMin)
+	}
+	if durationValue.Max != expectedDurationMax {
+		t.Errorf("got tag '%v' max %v, want %v", mutatorIngestionDurationMetricName, durationValue.Max, expectedDurationMax)
+	}
+
+	verifyTags(t, expectedTags, row.Tags)
+}
+
+func checkData(t *testing.T, name string, rowLength int) *view.Row {
+	row, err := view.RetrieveData(name)
+	if err != nil {
+		t.Errorf("Error when retrieving data: %v from %v", err, name)
```

Comment:
I changed this function to send back errors, rather than report directly using `t`.  So now that error short circuits the function and there's no opportunity for a panic.